Optics/Functorial.agda

```diff
@@ -50,6 +51,10 @@ module Optics.Functorial where
   set (lens p) a s = p id if (const a) s
   syntax set p a s = s [ p := a ]
 
+  set? : ∀{S A} → Lens S (Maybe A) → A → S → S
+  set? l a s = s [ l := just a ]
+  syntax set? l a s = s [ l ?= a ]
```

Comment:
Any reason not to call this operator `?~` instead of `?=`?  I think that's the Haskell spelling.  I'm also wondering whether we could/should do:
```
  syntax set? l a s = s & l ?~ a 
```
for better consistency with Haskell syntax.  (Similarly for `set` above.)  These aren't a huge detail, but the closer the model is to the implementation, the easier is too review, and the less risk there is of overlooking a difference.

Reply:
In Haskell, the `&` operator is function application. We I propose we rename `_|>_` from the Agda standard library's `Function` module (which does the same thing) to `_&_`, and define syntax for `l ?= a : S -> S`, to make the correspondence as close as possible.

Reply:
This might be useful:

[https://hackage.haskell.org/package/lens-5.0.1/docs/Control-Lens-Operators.html](https://hackage.haskell.org/package/lens-5.0.1/docs/Control-Lens-Operators.html)

Reply:
@cwjnkins that sounds reasonable to me, but I'm not going to hold up this PR for it.  It would be great if you could experiment with this and, if it makes sense, sneak it into another PR soon.

Reply:
Oh, never mind, I see you've already done it.  Great!